valohai_yaml/pipelines/conversion.py

```diff
@@ -111,6 +113,11 @@ def convert_executionlike_node(
                 step_data.pop("no-output-timeout"),
             )
 
+        # Shallow-copy the step before we mutate it,
+        # so the changes don't reflect into `self.config`
+        step = copy.copy(step)
+        step.inputs = OrderedDict((key, input.with_edge_merge_mode_applied()) for key, input in step.inputs.items())
+
```

Comment:
I think we should override inputs only when there is an actual edge configured. If no edge is configured to override this specific input it should use the defaults. If I read this correctly it's also clearing the defaults from inputs that are not part of an edge.

Reply:
Based on this and @akx and my discussion today. This feature cannot be done in valohai-yaml as a part of the logic depends on if there is an actual edge(s) and thus should be handled when the actual incoming edge data is resolved in the system actually managing the pipeline (Roi for example). Also the configuration can be altered after it's read from the config file and before it's send to a server.